mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractCombinationManagerTest.java

```diff
@@ -0,0 +1,284 @@
+
+package org.apache.james.mailbox.store;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+import java.util.Date;
+
+import javax.mail.Flags;
+import javax.mail.Flags.Flag;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.MessageManager.FlagsUpdateMode;
+import org.apache.james.mailbox.MessageManager.MetaData.FetchGroup;
+import org.apache.james.mailbox.manager.MailboxManagerFixture;
+import org.apache.james.mailbox.mock.MockMailboxSession;
+import org.apache.james.mailbox.model.ComposedMessageId;
+import org.apache.james.mailbox.model.FetchGroupImpl;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.model.MultimailboxesSearchQuery;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+public abstract class AbstractCombinationManagerTest {
+    private static final Flags FLAGS = new Flags();
+    private static final byte[] SUBJECT_TEST_MAIL = "Subject: test\r\n\r\ntestmail".getBytes();
+
+    private MailboxManager mailboxManager;
+    private MessageIdManager messageIdManager;
+    private MessageManager messageManager;
+
+    private MailboxSession session;
+    private Mailbox mailbox1;
+    private Mailbox mailbox2;
+
+    private CombinationManagerTestSystem testingData;
+
+    public abstract CombinationManagerTestSystem createTestingData() throws Exception ;
+
+    @Before
+    public void setUp() throws Exception {
+        session = new MockMailboxSession(MailboxManagerFixture.USER);
+        testingData = createTestingData();
+
+        mailbox1 = testingData.createMailbox(MailboxManagerFixture.MAILBOX_PATH1, session);
+        mailbox2 = testingData.createMailbox(MailboxManagerFixture.MAILBOX_PATH2, session);
+
+        mailboxManager = testingData.getMailboxManager();
+        messageIdManager = testingData.getMessageIdManager();
+        messageManager = testingData.createMessageManager(mailbox1, session);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        testingData.clean();
+    }
+
+    @Test
+    public void getMessageCountFromMessageManagerShouldReturnDataSetInMailboxesFromMessageIdManager() throws Exception {
+        MessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
+
+        assertThat(messageManager.getMessageCount(session)).isEqualTo(1);
```

Comment:
Nope. messageManager corresponds to mailbox1.

You should introduce a messageManager2 for mailbox2 and use it here to retrieve the count.